--- origsrc/xwininfo-1.1.2/configure.ac	2011-06-29 23:25:26.000000000 -0500
+++ src/xwininfo-1.1.2/configure.ac	2011-08-22 12:29:23.763640300 -0500
@@ -45,9 +45,7 @@ if test "x$ac_cv_func_strnlen_working" =
   AC_DEFINE(HAVE_STRNLEN, 1, [Define to 1 if you have a working strnlen function.])
 fi
 
-# Check for iconv in libc, then libiconv
-AC_SEARCH_LIBS([iconv], [iconv], [AC_DEFINE([HAVE_ICONV], 1,
-	[Define to 1 if you have the iconv() function])])
+AM_ICONV
 
 # Allow using xcb-icccm, but don't make it the default while the API is
 # still being changed.
@@ -70,6 +68,7 @@ PKG_CHECK_MODULES(XWININFO, [xcb >= 1.6]
 # and libX11 headers for cursorfont.h
 PKG_CHECK_MODULES(XLIB, x11 [xproto >= 7.0.17])
 XWININFO_CFLAGS="${XWININFO_CFLAGS} ${XLIB_CFLAGS}"
+XWININFO_LIBS="${XWININFO_LIBS} ${LIBICONV}"
 
 AC_CONFIG_FILES([
 	Makefile
--- origsrc/xwininfo-1.1.2/xwininfo.c	2011-06-18 12:20:51.000000000 -0500
+++ src/xwininfo-1.1.2/xwininfo.c	2011-08-22 12:36:04.167067400 -0500
@@ -1900,7 +1900,7 @@ print_utf8 (const char *prefix, char *u8
 
     if (iconv_from_utf8 != (iconv_t) -1) {
 	Bool done = True;
-	char *inp = u8str;
+	ICONV_CONST char *inp = u8str;
 	char convbuf[BUFSIZ];
 	int convres;
 
